apps/session-manager/src/utils/redis-trace-wrapper.ts

```diff
@@ -0,0 +1,84 @@
+import * as redis from "redis";
+import appInsights from "applicationinsights";
+
+function wrapAsyncFunctionWithAppInsights<
+  K extends keyof redis.RedisClusterType,
+  T extends redis.RedisClusterType[K],
+>(
+  redisClient: redis.RedisClusterType,
+  originalFunction: T,
+  functionName: string,
+  clientName: string,
+  appInsightsClient?: appInsights.TelemetryClient,
+): T {
+  return async function (...args: unknown[]) {
+    const startTime = Date.now();
+
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const result = await (originalFunction as any).apply(redisClient, args);
```

Comment:
This works even with sync methods of the Redis client? Wrapping only the COMMANDS should be fine. I'll check if it works only with legacy mode `off`.

Reply:
await a non-promise function is fine. But do you know how to filter commands only?

Reply:
We could use this code:
```typescript
import commands from "@redis/client/dist/lib/cluster/commands";

...

Object.keys(commands) // List of commands
```